R/plotCorrelations.R

```diff
@@ -0,0 +1,319 @@
+#' @param fileName (character)     if not NULL - it will use the Cairo package to create a
+#'                                 device. It checks the file extension to
+#'                                 instantiate appropriate device;
+#'                                 otherwise - it does nothing (graphs are plotted on screen)
+#' @return (NULL)
+#' @author \href{https://orcid.org/0000-0002-8778-6442}{Denitsa Vasileva} (aut)
+#' [Cairo package](https://CRAN.R-project.org/package=Cairo
+preDraw <- function(fileName) {
+  if (is.null(fileName)) {
+    return(invisible(NULL))
+  }
+  # is the file extension pdf?
+  if (grepl(".pdf$", fileName)) {
+    Cairo::CairoPDF(
+      file = fileName,
+      width = 1920,
+      height = 1080,
+      res = 92,
+      pointsize = 12
+    )
+    return(invisible(NULL))
+  }
+  # otherwise - default to png
+  if (!grepl(".png$", fileName)) {
+    fileName <- paste0(fileName, ".png")
+  }
+  # Cairo Package https://CRAN.R-project.org/package=Cairo
+  Cairo::CairoPNG(
+    file = fileName,
+    width = 1920,
+    height = 1080,
+    res = 92,
+    pointsize = 12
+  )
+  return(invisible(NULL))
+}
+
+#' Helper function for the co-expression correlation & semantic similarity correlation heat maps
+#' @param fileName - if not NULL - it will call dev.off to save the file
+#' otherwise - it prompts for user input - so the user can see the graph
+#' before the next one is drawn
+#' @return (NULL)
+#' @author \href{https://orcid.org/0000-0002-8778-6442}{Denitsa Vasileva} (aut)
+postDraw <- function(fileName) {
+  if (is.null(fileName)) {
+    readline(prompt = "Press enter to continue...")
+  } else {
+    dev.off()
+  }
+  return(invisible(NULL))
+}
+
+
+#'
+#' \code{installHumanGenomeAnnotation()} installs Human Genome Annotation
+#' as per Dr. Stipe's email and
+#' https://bioconductor.org/packages/release/data/annotation/html/org.Hs.eg.db.html
+#'
+installHumanGenomeAnnotation <- function() {
+  if (!requireNamespace("BiocManager", quietly = TRUE)) {
+    install.packages("BiocManager")
+  }
+  BiocManager::install("org.Hs.eg.db", version = "3.8")
+  return(invisible(NULL))
+}
+
+#'
+#' \code{computeCorrelations()} Calculates pairwise Semantic similarity and co-expression correlation
+#' between expression profiles of genes and returns a
+#' data frame
+#'
+#' @author \href{https://orcid.org/0000-0002-8778-6442}{Denitsa Vasileva} (aut)
+#' @param silent = FALSE suppresses all messages when TRUE
+#' @return (data frame)           containing gene A, gene B, correlation between them and
+#'                                GOSemSim correlation
+#' [GOSemSim package](http://bioconductor.org/packages/release/bioc/html/GOSemSim.html)
+#' [org.Hs.eg.db data package](https://bioconductor.org/packages/release/data/annotation/html/org.Hs.eg.db.html)
+computeCorrelations <- function(silent = FALSE) {
+  sysDB <- fetchData("SysDB")
+
+  # Dr.Steipe's data file
+  geoQNURL <- paste0(
```

Comment:
I really object to accessing this "sideways" when the function `fetchData()` exists exactly for this purpose. This is super brittle because the URL and the filename could change at any time.